### PR TITLE
Firebase auth

### PR DESCRIPTION
**Pull request by AhmedAdel999**
Added firebase auth to the project so now we use the token returned to us by firebase instead of dummy uuids.

### CODE REVIEW

File: modules/HomePage/pages/home-page.tsx [hunk 1]
```diff
@@ -13,6 +14,17 @@ interface WidgetData {
 }
 
 const HomePage: FC<IPostFeed.IPosts> = ({ data }): ReactElement => {
+  const router = useRouter();
+  useEffect(() => {
+    if (document.cookie.split('=')[0] !== 'user') {
```
Comment (Nilegfx) [inline modules/HomePage/pages/home-page.tsx:19]: too naive way. please use cookies APIs

File: modules/HomePage/pages/home-page.tsx [hunk 2]
```diff
@@ -13,6 +14,17 @@ interface WidgetData {
...
+      router
+        .push('/login')
+        .then()
```
Comment (Nilegfx) [inline modules/HomePage/pages/home-page.tsx:22]: we discussed this earlier. this is immature way of handling promises.

File: modules/HomePage/pages/home-page.tsx (hunk 1 above)
Comment (AhmedAdel999) [reply modules/HomePage/pages/home-page.tsx:19]: You mean a package like cookie-cutter?

File: modules/HomePage/pages/home-page.tsx (hunk 2 above)
Comment (AhmedAdel999) [reply modules/HomePage/pages/home-page.tsx:22]: I tried to `return router.push('/login')` but there was an error regarding useEffect, I will investigate it further.

File: modules/Login/pages/Login.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import type { FC, ReactElement } from 'react';
+import { useEffect } from 'react';
+import { useRouter } from 'next/router';
+import * as EButton from '../../shared/components/atoms/Button/types/EButton';
+import { loginUser, register } from '../../shared/api/auth';
+
+const Login: FC = (): ReactElement => {
+  const router = useRouter();
+  useEffect(() => {
+    if (document.cookie.split('=')[0] === 'user') {
```
Comment (Nilegfx) [inline modules/Login/pages/Login.tsx:11]: this logic "determining whether the user is logged in or not is repeated. we discussed that this should be managed by a custom hook for cleaner code (I am aware of Elhaw suggestion for later improvement but this still not good enough to be merged)

File: modules/Login/pages/Login.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      router
+        .push('/')
+        .then()
```
Comment (Nilegfx) [inline modules/Login/pages/Login.tsx:14]: same comment

File: modules/Login/pages/Login.tsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        .catch((err) => {
+          console.log(err);
+        });
+    }
+  }, []);
+
+  const login = async (): Promise<boolean> => {
+    loginUser();
+    const errorObj = await register();
```
Comment (Nilegfx) [inline modules/Login/pages/Login.tsx:23]: mmmm.. the fact that you named the variable errorObj indicates for me that you need to read more about promises and async operations.

File: modules/HomePage/pages/home-page.tsx (hunk 1 above)
Comment (Nilegfx) [reply modules/HomePage/pages/home-page.tsx:19]: have you tried to google "cookies API" before asking this question? 😄

File: modules/Login/pages/Login.tsx [hunk 6]
```diff
@@ -19,12 +19,12 @@ const Login: FC = (): ReactElement => {
 
   const login = async (): Promise<void> => {
     await loginUser();
-    const isError = await register();
-    if (!isError) {
+    const { resData } = await register();
```
Comment (AhmedAdel999) [inline modules/Login/pages/Login.tsx:22]: Why did you change that?

**Review by Nilegfx — APPROVED** [review]
LGTM (didn't review it but to unblock you)
